src/pages/cart.jsx

```diff
@@ -23,13 +23,97 @@ import {
 import { Seo } from "../components/seo"
 
 export default function CartPage() {
-  const { checkout, loading } = React.useContext(StoreContext)
+  const { checkout, loading, client, setCheckout } = React.useContext(StoreContext)
   const emptyCart = checkout.lineItems.length === 0
 
-  const handleCheckout = () => {
+  const handleCheckout = async () => {
+    if(window.onwardApp) {
+      await window.onwardApp.updateCartForCheckout();
+    }
     window.open(checkout.webUrl)
   }
 
+  const parseCheckout = (checkout) => {
+    return {
+      currency: checkout.currencyCode || null,
+      items: checkout.lineItems.map(item =>
+        ({
+          id: item.id,
+          cost: { amount: Number(item.variant.price.amount) * item.quantity, currencyCode: item.variant.price.currencyCode },
+          quantity: item.quantity,
+          properties: null,
+          variant: {
+            price: null,
+            requiresShipping: true,
```

Comment:
the default buy button client doesn't include this so I've just hard coded for the example, you can change the graphql queries but didn't seem worth it for this example.